src/world.rs

```diff
@@ -68,7 +68,7 @@ impl World {
     }
 
     #[allow(unreachable_code)]
-    pub fn handle_player(&self, mut stream: TcpStream) -> io::Result<()> {
+    pub fn handle_player(&self, mut stream: SymmStream) -> io::Result<()> {
```

Comment:
That will be difficult to merge with your single-player branch, because of [the `try_clone` call](https://github.com/toqueteos/hematite_server/blob/cf110e63ef57be1049636e95dace6afd16b9980b/src/world.rs#L188).

I was thinking about introducing a `TryClone` trait and implementing it for the `SymmStream`, plus an additional `PlainStream` which would do nothing but wrap the underlying `TcpStream`.